### PR TITLE
Fix notification shows artifact vector drawable issue #1952

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Fix issue #1952:  Android M shows roundabout maneuver icon artifact in the foreground notification.
The problem was we should use AppCompatDrawable for vectors on M or prior to M.
We updated the maneuver icon resources for the notification's RemoteView. When it's a vector drawable and as @Guardiola31337 mentioned in #1952 that some Arc curves icon will show artifact.

The way this PR fix it is: get the AppCompatDrawable and create a bitmap for the maneuver icon then apply to the notification's RemoteView to update.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Screenshots or Gifs

**Before fix on Nexus 5 OS 6**
before fix on Nexus 5 OS 6

**After fix on Nexus 5 OS 6**
after fix on Nexus 5 OS 6

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by JunDai** [comment]
this PR replaces the origin one #2476

**Comment by Guardiola31337** [comment]
@JunDai CI is ❌ 

A bunch of tests failed 👀 

`build`

```
455 tests completed, 44 failed, 1 skipped
```

`build-1.0`

```
11 tests completed, 7 failed
```

Could you take a look at those?

**Comment by JunDai** [comment]
@Guardiola31337 - unit tests failure is fixed, would you mind to take a look again?

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java [hunk 1]
```diff
@@ -592,8 +594,12 @@ private MapboxNavigationNotification buildMapboxNavigationNotification() {
     MapboxNavigation mockedMapboxNavigation = createMapboxNavigation();
     Context mockedContext = createContext();
     Notification mockedNotification = mock(Notification.class);
-    return new MapboxNavigationNotification(mockedContext,
-      mockedMapboxNavigation, mockedNotification);
+    MapboxNavigationNotification notification = new MapboxNavigationNotification(mockedContext,
+            mockedMapboxNavigation, mockedNotification);
+
+    final MapboxNavigationNotification spyNotification = Mockito.spy(notification);
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java]: NIT

```suggestion
    MapboxNavigationNotification spyNotification = Mockito.spy(notification);
```

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java [hunk 2]
```diff
@@ -592,8 +594,12 @@ private MapboxNavigationNotification buildMapboxNavigationNotification() {
...
+    Mockito.doReturn(null).when(spyNotification).getManeuverBitmap(anyInt());
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java]: NIT

```suggestion
    doReturn(null).when(spyNotification).getManeuverBitmap(anyInt());
```

File: libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationTest.kt [hunk 3]
```diff
@@ -313,6 +314,10 @@ class MapboxTripNotificationTest {
         mockkStatic(TextUtils::class)
         val slot = slot<CharSequence>()
         every { TextUtils.isEmpty(capture(slot)) } answers { slot.captured.isEmpty() }
+
+        mockkStatic(AppCompatResources::class)
```
Comment (Guardiola31337) [inline libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationTest.kt]: Should we do `unmockkObject` afterwards?

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java (hunk 1 above)
Comment (JunDai) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java]: done

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java (hunk 2 above)
Comment (JunDai) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotificationManeuverResourceTest.java]: done

**Pull request merged**